exporter/otlp/test/opentelemetry/exporter/otlp/exporter_test.rb

```diff
@@ -148,31 +148,68 @@
         _(exp.instance_variable_get(:@headers)).must_equal('token' => 'über')
       end
 
-      it 'omits any invalid header values' do
-        logger = OpenTelemetry.logger
-        log_stream = StringIO.new
-        OpenTelemetry.logger = ::Logger.new(log_stream)
+      it 'fails fast when header values are missing' do
+        error = _(){
+                     with_env('OTEL_EXPORTER_OTLP_HEADERS' => 'a = ') do
+                       OpenTelemetry::Exporter::OTLP::Exporter.new
+                     end
+                   }.must_raise(ArgumentError)
+        _(error.message).must_match(/headers/i)
 
-        exp = with_env('OTEL_EXPORTER_OTLP_HEADERS' => 'a=,c=hi%F3,,token=abc455==afs') do
-          OpenTelemetry::Exporter::OTLP::Exporter.new
-        end
-        _(exp.instance_variable_get(:@headers)).must_equal('token' => 'abc455==afs')
+        error = _(){
+                     with_env('OTEL_EXPORTER_OTLP_TRACES_HEADERS' => 'a = ') do
```

Comment:
I'd like to clean up all these tests at some point, to remove the duplication between `OTEL_EXPORTER_OTLP_` and `OTEL_EXPORTER_OTLP_TRACES_` prefixes. It's pretty pervasive in this file, so let's do that in a subsequent PR rather than polluting this one.

Reply:
Sounds good. I don't often use minitest spec format and teams usually have very specific style rules about how they prefer test layout. It occurs now to me that I don't know what the maintainers preferred style is here. 

E.g. do y'all prefer using a `describe` block per variation and an `it` block per example?